### PR TITLE
[NO CHANGELOG] [Add Funds Widget] Get token to calculate USD amount

### PR DESCRIPTION
**Pull request by jiyounglee**
Get token to calculate USD amount

### CODE REVIEW

**Review by mimi-imtbl — APPROVED** [review]

File: packages/checkout/widgets-lib/src/widgets/add-funds/views/AddFunds.tsx [hunk 1]
```diff
@@ -230,6 +231,31 @@ export function AddFunds({
   //   console.log('handle review click');
   // }, []);
 
+  useEffect(() => {
```
Comment (jhesgodi) [inline packages/checkout/widgets-lib/src/widgets/add-funds/views/AddFunds.tsx:234]: Doing this calculation in react, reduces the testability and performance of the code.

I suggest you move these to a `getFiatAmount` utility fn:

```export function getFiatAmount(amount: string, token: TokenInfo, tokens: TokenInfo[]): Number {
  if (!amount || !token?.address || !tokens) {
    return 0;
  }

  const address =
    token.address === "native" ? SQUID_NATIVE_TOKEN : token?.address;

  const toToken = tokens.find(
    (value) =>
      value.address.toLowerCase() === address.toLowerCase() &&
      value.chainId === ChainId.IMTBL_ZKEVM_MAINNET.toString()
  );

  if (!toToken) {
    return 0;
  }

  return Number(amount) * toToken.usdPrice;
}
```
and set the value using a `useMemo` hook.

```
const fiatAmount = useMemo(
  () => getFiatAmount(inputValue, currentToTokenAddress, tokens);
}, [inputValue, currentToTokenAddress, tokens]);
```

File: packages/checkout/widgets-lib/src/widgets/add-funds/views/AddFunds.tsx [hunk 2]
```diff
@@ -230,6 +231,31 @@ export function AddFunds({
...
+        && value.chainId === ChainId.IMTBL_ZKEVM_MAINNET.toString(),
+    );
+
+    if (!toToken) {
+      return;
+    }
+
+    const tokenBalance = Number(inputValue) * toToken.usdPrice;
+    if (tokenBalance > 0) {
+      setAmountInUSD(`USD $${tokenBalance.toFixed(2)}`);
```
Comment (jhesgodi) [inline packages/checkout/widgets-lib/src/widgets/add-funds/views/AddFunds.tsx:255]: Please avoid `USD $${tokenBalance.toFixed(2)}` as the amounts format will be added by the translation interpolation.

ie:
`t(`ADD_FUNDS_FIAT_AMOUNT`, { fiatAmount })` where `amount` is `number | string`

this allows the language to set the format for the currency template, also notice rename it to `fiatAmount` as anchor currency could change

**Review by jhesgodi — CHANGES_REQUESTED** [review]
review it with some comments

**Review by jhesgodi — APPROVED** [review]

File: packages/checkout/widgets-lib/src/widgets/add-funds/views/AddFunds.tsx (hunk 1 above)
Comment (jiyounglee) [reply packages/checkout/widgets-lib/src/widgets/add-funds/views/AddFunds.tsx]: Done

File: packages/checkout/widgets-lib/src/widgets/add-funds/views/AddFunds.tsx (hunk 2 above)
Comment (jiyounglee) [reply packages/checkout/widgets-lib/src/widgets/add-funds/views/AddFunds.tsx]: Done

**Review by mimi-imtbl — DISMISSED** [review]

**Review by mimi-imtbl — APPROVED** [review]

**Pull request merged**